great_expectations/data_context/data_context.py

```diff
@@ -1223,7 +1239,7 @@ def _get_data_context_version(self, arg1: Any, **kwargs) -> Optional[str]:
         Returns "v2" if the datasource is an instance of the LegacyDatasource class.
         """
 
-        if not self.datasources:
+        if self.datasources is {}:
```

Comment:
The previous version `if not self.datasources` is a stronger check.

Reply:
you're right. will change back :)